notify/wecomrobot/wecomrobot.go

```diff
@@ -0,0 +1,88 @@
+package wecomrobot
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+	commoncfg "github.com/prometheus/common/config"
+)
+
+// Notifier implements a Notifier for generic wecomrobot.
+type Notifier struct {
+	conf   *config.WeComRobotConfig
+	tmpl   *template.Template
+	logger log.Logger
+	client *http.Client
+}
+
+// New returns a new Wechat notifier.
+func New(c *config.WeComRobotConfig, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "wecomrobot", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Notifier{conf: c, tmpl: t, logger: l, client: client}, nil
+}
+
+type Mark struct {
+	Content string `json:"content"`
+}
+
+type WeComRobotMessage struct {
+	Msgtype string `json:"msgtype"`
+	Text    Mark   `json:"text"`
+}
+
+// Notify implements the Notifier interface.
+func (n *Notifier) Notify(ctx context.Context, alerts ...*types.Alert) (bool, error) {
+	fmt.Println("alert:", alerts[0].Labels)
+
+	var (
+		tmplErr error
+		data    = notify.GetTemplateData(ctx, n.tmpl, alerts, n.logger)
+		tmpl    = notify.TmplText(n.tmpl, data, &tmplErr)
+	)
+	if tmplErr != nil {
+		return false, fmt.Errorf("templating error: %s", tmplErr)
+	}
+
+	message := tmpl(n.conf.Message)
+	if tmplErr != nil {
+		return false, fmt.Errorf("templating error: %s", tmplErr)
+	}
+
+	msg := WeComRobotMessage{
+		Msgtype: "text",
+		Text:    Mark{Content: message},
+	}
+
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
+		return false, err
+	}
+
+	resp, err := notify.PostJSON(ctx, n.client, n.conf.WebhookURL.String(), &buf)
+	if err != nil {
+		return true, err
+	}
+	defer notify.Drain(resp)
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return true, err
+	}
+	level.Debug(n.logger).Log("response", string(body))
+
+	return true, nil
```

Comment:
Is it necessary to parse fields of resp.Body to check success or failure of internal business logic?